scripts/build.sh

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env bash
+
+set -eu
+set -o pipefail
+
+readonly PROGDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+readonly BUILDERDIR="$(cd "${PROGDIR}/.." && pwd)"
+readonly LOCAL_REGISTRY_NAME="builder_test_registry"
+
+# shellcheck source=SCRIPTDIR/.util/tools.sh
+source "${PROGDIR}/.util/tools.sh"
+
+# shellcheck source=SCRIPTDIR/.util/print.sh
+source "${PROGDIR}/.util/print.sh"
+
+function main() {
+  local name token
+  token=""
+
+  while [[ "${#}" != 0 ]]; do
+    case "${1}" in
+      --help|-h)
+        shift 1
+        usage
+        exit 0
+        ;;
+
+      --name|-n)
+        name="${2}"
+        shift 2
+        ;;
+
+      --token|-t)
+        token="${2}"
+        shift 2
+        ;;
+
+      "")
+        # skip if the argument is empty
+        shift 1
+        ;;
+
+      *)
+        util::print::error "unknown argument \"${1}\""
+    esac
+  done
+
+  if [[ -z "${name:-}" ]]; then
+    name="testbuilder"
+  fi
+
+  # tools::install "${token}"
+  if [[ ! -f "${BUILDERDIR}/.bin/pack" ]]; then
+    experimental::pack::install "${token}"
+  fi
+
+  # Do not rebuild the builder if it already exists on the local registry
+  if [[  -z $(docker ps -aqf "name=${LOCAL_REGISTRY_NAME}") ]]; then
+    local_registry::run >/dev/null
+  else
+    echo "Local registry named ${LOCAL_REGISTRY_NAME} is already running"
+    registry_images=$(curl -X GET "localhost:5000/v2/_catalog" | jq -r '.repositories')
+    if echo "${registry_images}" | grep -q "${name}"; then
+      echo "A builder image named localhost:5000/${name} on the local registry already exists"
+      exit 0
+    fi
+  fi
+
+  name="localhost:5000/${name}"
+  builder::create "${name}"
+}
+
+function usage() {
+  cat <<-USAGE
+build.sh [OPTIONS]
+
+Builds the multi-arch builder using a local docker registry
+
+OPTIONS
+  --help        -h         prints the command usage
+  --name <name> -n <name>  sets the name of the builder that is built
+  --token <token>          Token used to download assets from GitHub (e.g. jam, pack, etc) (optional)
+USAGE
+}
+
+function tools::install() {
+  local token
+  token="${1}"
+
+  util::tools::pack::install \
+    --directory "${BUILDERDIR}/.bin" \
+    --token "${token}"
+}
+
+# TODO: when an official pack release comes out that supports multi-arch, use that
+#  Until then, use an experimental pack version we've saved to a shared location
+function experimental::pack::install() {
+  echo "Installing pack experimental"
+
+  local artifact_uri
+  os=$(util::tools::os)
+  arch=$(util::tools::arch)
+  mkdir -p "${BUILDERDIR}/.bin"
+
+  if [[ ${arch}  == "amd64" ]]; then
+    if [[ ${os}  == "linux" ]]; then
+      # linux + amd64
+      artifact_uri="https://api.github.com/repos/buildpacks/pack/actions/artifacts/1306592515/zip"
+    elif [[ ${os}  == "darwin" ]]; then
+      # darwin + amd64
+      artifact_uri="https://api.github.com/repos/buildpacks/pack/actions/artifacts/1306588326/zip"
```

Comment:
If I recall correctly these links will break after 30 days or so. Do we have more permanent links to use?

Reply:
we can put it on box.com too?

Reply:
I would just prefer that this did not break suddenly because the link died

Reply:
Right, @ForestEckhardt I'm asking if you know of a good Paketo project storage location to avoid this!

Reply:
Besides that there is no "permanent" link to use because the CLI is being iterated on before release

Reply:
I think it might make sense if this is a temporary solution to just vendor the binaries into the repository itself. Just put them in the scripts folder and we can remove them once the workflow is actually supported.

Reply:
let me try that

Reply:
https://github.com/paketo-buildpacks/builder-jammy-buildpackless-tiny/pull/203/commits/bb5ffc9199953a930d3148cd4f7b4536a7967b15